tests/components/gios/test_config_flow.py

```diff
@@ -1,138 +1,93 @@
 
 
-async def test_form_with_api_error(hass: HomeAssistant) -> None:
+async def test_form_with_api_error(hass: HomeAssistant, mock_gios: MagicMock) -> None:
     """Test the form is aborted because of API error."""
-    with patch(
-        "homeassistant.components.gios.coordinator.Gios._get_stations",
-        side_effect=ApiError("error"),
-    ):
-        result = await hass.config_entries.flow.async_init(
-            DOMAIN, context={"source": SOURCE_USER}
-        )
+    mock_gios.create.side_effect = ApiError("error")
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
 
     assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "cannot_connect"
 
 
-async def test_invalid_sensor_data(hass: HomeAssistant) -> None:
+async def test_invalid_sensor_data(hass: HomeAssistant, mock_gios: MagicMock) -> None:
     """Test that errors are shown when sensor data is invalid."""
-    with (
-        patch(
-            "homeassistant.components.gios.coordinator.Gios._get_stations",
-            return_value=STATIONS,
-        ),
-        patch(
-            "homeassistant.components.gios.coordinator.Gios._get_station",
-            return_value=json.loads(
-                await async_load_fixture(hass, "station.json", DOMAIN)
-            ),
-        ),
-        patch(
-            "homeassistant.components.gios.coordinator.Gios._get_sensor",
-            return_value={},
-        ),
-    ):
-        flow = config_flow.GiosFlowHandler()
-        flow.hass = hass
-        flow.context = {}
-
-        result = await flow.async_step_user(user_input=CONFIG)
-
-        assert result["errors"] == {CONF_STATION_ID: "invalid_sensors_data"}
-
-
-async def test_cannot_connect(hass: HomeAssistant) -> None:
-    """Test that errors are shown when cannot connect to GIOS server."""
-    with (
-        patch(
-            "homeassistant.components.gios.coordinator.Gios._get_stations",
-            return_value=STATIONS,
-        ),
-        patch(
-            "homeassistant.components.gios.coordinator.Gios._async_get",
-            side_effect=ApiError("error"),
-        ),
-    ):
-        result = await hass.config_entries.flow.async_init(
-            DOMAIN, context={"source": SOURCE_USER}
-        )
-        result = await hass.config_entries.flow.async_configure(
-            result["flow_id"], CONFIG
-        )
-        await hass.async_block_till_done()
+    mock_gios.create.return_value.async_update.side_effect = InvalidSensorsDataError(
+        "Invalid data"
+    )
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+    )
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input=CONFIG
+    )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["errors"] == {CONF_STATION_ID: "invalid_sensors_data"}
```

Comment:
This doesn't look good to me. If the station information couldn't be retrieved, there's nothing the user can do and we should close the form.

Is it related to https://github.com/home-assistant/core/pull/155603/files#r2483768155?

Reply:
Fixed.